### ISSUE TITLE
Changes in note titles for transcluded documents are not updated

### ISSUE THREAD

**Issue by charleshan** [zsviczian/obsidian-excalidraw-plugin#517]
**Describe the bug**
Changes in note titles for transcluded documents are not updated.

**To Reproduce**
1. Create an excalidraw file
2. Create a markdown file called "Test A"
3. Embed the markdown file in the excalidraw file using `![[Test A]]`
4. Change "Test A" to "Test B"

**Expected behavior**
The excalidraw file should update the embedded text to `![[Test B]]`

**Environment (please complete the following information):**
 - OS including version: macOS 12.2.1
 - Plugin version: 1.6.17
 - Obsidian version: 0.13.31

**Comment by zsviczian** [comment]
I can’t reproduce. Do you have “Automatically update internal links” enabled under Obsidian/Files and Links setting?

**Comment by charleshan** [comment]
Yes. Are you sure you tried `![[filename]]` and not `[[filename]]`? 

`[[filename]]` updates automatically. `![[filename]]` doesn't.

**Comment by zsviczian** [comment]
If you open the markdown view for the drawing you will see that the two (link & transclusion) are handled exactly the same. They both appear under text elements. The key reason for the markdown file format and for having text elements in the markdown file is so that Obsidian takes care of file renaming. The other reason is so you can reference text on a drawing in documents...

https://user-images.githubusercontent.com/14358394/157819644-192211ae-19b9-4661-b594-c8de0be06104.mp4

**Comment by charleshan** [comment]
Oh... It seems to be a different problem. I always work in multiple panes and I noticed that excalidraw doesn't get updated automatically so I've been toggling in between "Change to Raw mode" to fetch the latest changes. Now I see that excalidraw is overwriting the file with the old data.

https://user-images.githubusercontent.com/3143154/157927336-4c4d31f6-f1a2-4670-a684-f98464ef2061.mp4
